spyder/plugins/editor/widgets/base.py

```diff
@@ -681,6 +681,50 @@ def __move_line_or_selection(self, after_current_line=True):
         start_pos, end_pos = self.__save_selection()
         last_line = False
 
+        # Unfold any folded code block before moving lines up/down
+        folding_panel = self.panels.get('FoldingPanel')

```

Comment:
This class (`TextEditBaseWidget`) is shared between the editor and the Internal console. And the Internal console doesn't have panels.

So I'd suggest to redefine `__move_line_or_selection` in `CodeEditor` because all the code the you added only applies to it. Of course, you can call `super().__move_line_or_selection()` there to call what you need from this class.